rs/src/accounts_store.rs

```diff
@@ -1066,9 +1076,19 @@ impl AccountsStore {
     }
 
     fn is_create_canister_transaction(memo: Memo, to: &AccountIdentifier, principal: &PrincipalId) -> bool {
+        // There are now 2 ways to create a canister.
+        // The new way involves sending ICP directly to the CMC, the NNS Dapp canister then
+        // notifies the CMC of the transfer.
+        // The old way involves sending ICP to the NNS Dapp, the NNS Dapp then forwards the ICP
+        // on to the CMC and calls notify on the ledger which in turns notifies the CMC.
         if memo == MEMO_CREATE_CANISTER {
-            let expected_to = AccountIdentifier::new(dfn_core::api::id().get(), Some(principal.into()));
-            if *to == expected_to {
+            let subaccount = principal.into();
+            let expected_to_new = AccountIdentifier::new(CYCLES_MINTING_CANISTER_ID.into(), Some(subaccount));
+            if *to == expected_to_new {
+                return true;
+            }
```

Comment:
Please keep the scope the variables small, so that it is clear that let_expected_to is not used further down in the code.
```suggestion
           { // Sent to the CMC subaccount for this principal?
              let expected_to = AccountIdentifier::new(CYCLES_MINTING_CANISTER_ID.into(), Some(subaccount));
              if *to == expected_to {
                  return true;
              }
            }
```

Reply:
Done